applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdbool.h>
+
+#include "cloud_codec.h"
+#include <cJSON_os.h>
+#include <errno.h>
+
+struct cloud_codec_data codec;
+int ret;
+
+void setUp(void)
+{
+	ret = cloud_codec_init(NULL, NULL);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	memset(&codec, 0, sizeof(codec));
+}
+
+void tearDown(void)
+{
+	cJSON_FreeString(codec.buf);
+}
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+#define CONF_RECV_EXAMPLE \
```

Comment:
Can you explain how these are macros are defined? Do they correspond to some test data?

Reply:
these macros correspond to test data defined further below. they mainly act as examples on how the string conversion works.